lib/src/ui/pages/location/location_notifier.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:allthenews/src/domain/location/get_location_use_case.dart';
+import 'package:allthenews/src/domain/location/location.dart';
+import 'package:allthenews/src/domain/location/location_exception.dart';
+import 'package:allthenews/src/domain/location/location_resolver.dart';
+import 'package:allthenews/src/ui/pages/location/location_error_view_entity_mapper.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:geolocator/geolocator.dart';
+
+class LocationNotifier extends ChangeNotifier {
+  final GetLocationUseCase _getLocationUseCase;
+  final LocationResolver _locationResolver;
+  final LocationErrorViewEntityMapper _errorViewEntityMapper;
+
+  LocationViewState _state = const LocationViewState();
+
+  LocationNotifier(this._getLocationUseCase, this._errorViewEntityMapper, this._locationResolver);
+
+  LocationViewState get state => _state;
+
+  Future<void> findUserLocation() async {
+    _setNotifierState(const LocationViewState(isLoading: true));
+    await _getLocationUseCase()
+        .then((location) => _setNotifierState(LocationViewState(location: location)))
+        .catchError((exception) => _setNotifierState(
+            LocationViewState(error: _errorViewEntityMapper.map(exception as LocationException))));
```

Comment:
zastanawiam się nad tym podejściem z tworzeniem nowego obiektu za każdym razem kiedy coś się zmienia. Dlaczego tak jest? W ten sposób tracimy wszystkie inne informacje, nie lepiej mieć jeden obiekt stanu i na nim bazować i zmieniać tylko to co jest w tej chwili konieczne, a nie budować wszystko od nowa?

Reply:
robiąc copyWith też tworzysz za każdym razem nowy obiekt, po prostu przekopiowujesz resztę pól :P Tutaj tak naprawdę zmieniam to co jest konieczne, po prostu nie mam stanu w którym mam ustawioną więcej niż 1 property naraz, więc nie ma sensu żebym robił copyWith(isLoading: false, location: null, exceptionViewEntity: viewEntity) skoro mogę po prostu zrobić tak jak powyżej :)